.github/workflows/argocd-chart-update.yaml

```diff
@@ -0,0 +1,126 @@
+name: argocd chart update
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+
+on:
+  schedule:
+    - cron: "0 0 * * 1" # 00:00 on Mondays
+  workflow_dispatch:
+
+jobs:
+  update-embedded-argocd:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      id-token: write
+      pull-requests: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v6
+        with:
+          fetch-depth: 1
+          sparse-checkout: |
+            cmd
+            go.mod
+            go.sum
+            internal
+            pkg
+            Taskfile.yml
+
+      - name: Check for new ArgoCD Chart version
+        id: diff
+        run: |
+          set -euo pipefail
+
+          cd cmd/yokecd-installer/argocd
+          rm -f argo-cd-*.tgz
+
+          # https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/Chart.yaml
+          CHART_INDEX_URL="https://argoproj.github.io/argo-helm/index.yaml"
+          CHART_ENTRY="$(curl -fsSL "$CHART_INDEX_URL" | yq '.entries."argo-cd"[0]')"
+          CHART_URL="$(echo "$CHART_ENTRY" | yq -r '.urls[0]')"
+          curl -fsSLO "$CHART_URL"
+
+          if ! git diff --quiet; then
+            CHART_VERSION="$(echo "$CHART_ENTRY" | yq -r '.version')"
+            echo "new version detected (${CHART_VERSION}), opening PR..."
+            echo "version=$CHART_VERSION" >> "$GITHUB_OUTPUT"
+            echo "continue=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "ArgoCD Chart is up to date, no action needed, exiting..."
+          fi
+
+      - uses: actions/setup-go@v6
+        if: steps.diff.outputs.continue == 'true'
+        with:
+          go-version-file: go.mod
+
+      - name: Setup k3d
+        if: steps.diff.outputs.continue == 'true'
+        run: |
+          curl -s https://raw.githubusercontent.com/k3d-io/k3d/main/install.sh | bash
+          k3d cluster create ci --wait
```

Comment:
Nice. I've been thinking of trying `k3d`... Yoke generally uses `kind`. This gives me an excuse to dive deeper into it.